File: 12-class-photos.js
function classPhotos(redShirts, blueShirts) {
  // sort student heights, logn each = n
  redShirts.sort((a, b) => b - a);
  blueShirts.sort((a, b) => b - a);

  console.log(redShirts, blueShirts);
  const shirtColorInFirstRow = redShirts[0] < blueShirts[0] ? 'RED' : 'BLUE';

  for (let i = 0; i < redShirts.length; i++) {
    const redShirt = redShirts[i];
    const blueShirt = blueShirts[i];

    if (shirtColorInFirstRow === 'RED') {
      if (redShirt >= blueShirt) return false;
    } else if (blueShirt >= redShirt) return false;
  }

  return true;
}

console.log(classPhotos([1, 2, 3, 4, 8], [2, 3, 4, 5, 6]));

File: algo-expert/12-class-photos.py
def classPhotos(redShirtHeights, blueShirtHeights):
    redShirtHeights.sort(reverse=True)
    blueShirtHeights.sort(reverse=True)

    shirtColorInFirstRow = 'RED' if redShirtHeights[0] < blueShirtHeights[0] else 'BLUE'
    for index in range(len(redShirtHeights)):
        redShirtHeight = redShirtHeights[index]
        blueShirtHeight = blueShirtHeights[index]

        if shirtColorInFirstRow == 'RED':
            if redShirtHeight >= blueShirtHeight:
                return False
        else:
            if blueShirtHeight >= redShirtHeight:
                return False

    return True


print(classPhotos([1, 2, 3, 4, 5, 6, ], [2, 3, 4, 5, 6, 7, 8]))
